d2graph/d2graph.go

```diff
@@ -1052,6 +1052,35 @@ func (obj *Object) GetDefaultSize(mtexts []*d2target.MText, ruler *textmeasure.R
 	return &dims, nil
 }
 
+func (obj *Object) SizeToContent(contentBox *geo.Box, paddingX, paddingY, desiredWidth, desiredHeight float64, labelDims d2target.TextDimensions) shape.Shape {
```

Comment:
makes no sense for this to return a shape. please refrain from this pattern of having functions too tightly coupled with the calling code, been seeing it here and there.